### PR TITLE
Small fix to Theano softmax for numerical stability

### PR DESCRIPTION
**Pull request by srjoglekar246**
Reference: "Numerical Stability" part @ http://cs231n.github.io/linear-classify/#softmax

### CODE REVIEW

**Comment by srjoglekar246** [comment]
@farizrahman4u FYI

File: keras/backend/theano_backend.py [hunk 1]
```diff
@@ -1554,7 +1554,8 @@ def relu(x, alpha=0., max_value=None):
 def softmax(x, axis=-1):
     if axis == -1 or axis == x.ndim - 1:
         return T.nnet.softmax(x)
-    return T.exp(x) / T.exp(x).sum(axis=axis, keepdims=True)
+    return T.exp(x - x.max()) / T.exp(x - x.max()
+                                      ).sum(axis=axis, keepdims=True)
```
Comment (fchollet) [inline keras/backend/theano_backend.py]: Style nit: consider breaking the line at `/` (with parens), or after opening a parens

**Comment by bobchennan** [comment]
I think it is better to consider the max for each vector:
```
xm = x.max(axis=axis,keepdims=True)
return T.exp(x - xm) / T.exp(x - xm).sum(axis=axis, keepdims=True)
```
